### PR TITLE
add 'swapEndian' intrinsics to bitops & endians module . 

### PR DESCRIPTION
**Pull request by Parashurama**
EDIT: The failing test is caused by differences in type conversion between the different platforms & the VM.
Will go back to this later, once this issue is solved.
EDIT: needs integer cast support in VM (PR #5908 waiting for review)

### CODE REVIEW

**Comment by ephja** [comment]
I wouldn't mind losing the `Int` suffix. Also, is it necessary to export the procs for the individual types in addition to:
```nim
proc swapEndian*[T:SomeInteger](x: T): T
```
?

**Comment by Parashurama** [comment]
I think that the Int suffix was to prevent name confusion (no name collision) with
littleEndian/bigEndian enums in system.nim

**Comment by data-man** [comment]
@Parashurama 

Any progress?

**Comment by Parashurama** [comment]
issue #7587 as well as any sign extension bug should now be fixed.

File: lib/pure/bitops.nim [hunk 1]
```diff
@@ -32,6 +32,22 @@ const useICC_builtins = defined(icc) and useBuiltins
 const useVCC_builtins = defined(vcc) and useBuiltins
 const arch64 = sizeof(int) == 8
 
+template toUint32[T](x: T): uint32 =
+  when sizeof(x) == 1: cast[uint8](x).uint32
+  elif sizeof(x) == 2: cast[uint16](x).uint32
+  else:                cast[uint32](x)
```
Comment (Araq) [inline lib/pure/bitops.nim]: I think this should be ``elif sizeof(x) == 4: cast[uint32](x) else: {.error: "invalid size".}``  and likewise for the other parts where this pattern is used.

**Comment by krux02** [comment]
we already have https://github.com/nim-lang/Nim/blob/devel/lib/pure/endians.nim

File: lib/pure/bitops.nim (hunk 1 above)
Comment (timotheecour) [reply lib/pure/bitops.nim:38]: not relevant as of 1.3.5 278b4582f5c7cc3471bccc4ba587030edf7a50e2

Comment (timotheecour) [reply lib/pure/bitops.nim:38]: oups, didn't realize PR was closed but not merged

**Pull request closed without merging**